Adding new indirect selection pages

runleonarun opened the pull request:
closes #4773 

## What are you changing in this pull request and why?
<!---
Describe your changes and why you're making them. If related to an open 
issue or a pull request on dbt Core, then link to them here! 

To learn more about the writing conventions used in the dbt Labs docs, see the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md).
-->

## Checklist
<!--
Uncomment when publishing docs for a prerelease version of dbt:
- [ ] Add versioning components, as described in [Versioning Docs](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-entire-pages)
- [ ] Add a note to the prerelease version [Migration Guide](https://github.com/dbt-labs/docs.getdbt.com/tree/current/website/docs/docs/dbt-versions/core-upgrade)
-->
- [ ] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) so my content adheres to these guidelines.
- [ ] For [docs versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#about-versioning), review how to [version a whole page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) and [version a block of content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content).
- [ ] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

Adding or removing pages (delete if not applicable):
- [ ] Add/remove page in `website/sidebars.js`
- [ ] Provide a unique filename for new pages
- [ ] Add an entry for deleted pages in `website/static/_redirects`
- [ ] Run link testing locally with `npm run build` to update the links that point to deleted pages

runleonarun committed:
addinb new indirect selection pages

runleonarun committed:
Merge branch 'current' into indirect-selection4773

runleonarun committed:
adding to sidebar

runleonarun committed:
closing file

runleonarun committed:
Merge branch 'indirect-selection4773' of github.com:dbt-labs/docs.getdbt.com into indirect-selection4773

runleonarun committed:
fixing the env var example

runleonarun committed:
Merge branch 'current' into indirect-selection4773

runleonarun commented on website/docs/reference/global-configs/indirect-selection.md:
```diff
@@ -0,0 +1,50 @@
+
+Options are `empty`, `buildable`, `cautious`, and `eager` (default). By default, dbt indirectly selects all tests if they touch any resource you select. Read more about these options in [Indirect selection in Test selection examples](/reference/node-selection/test-selection-examples?indirect-selection-mode=empty#indirect-selection).
+
+For example, you can run tests that only refer to selected nodes using a CLI configuration:
```
```suggestion
You can run tests that only refer to selected nodes using a CLI configuration, for example:
```

runleonarun commented on website/docs/reference/global-configs/indirect-selection.md:
```diff
@@ -0,0 +1,50 @@
+
+</File>
+
+For example, you can run tests that only refer to selected nodes using `profiles.yml` user configurations:
```
```suggestion
You can run tests that only refer to selected nodes using `profiles.yml` user configurations, for example:
```

runleonarun commented on website/docs/reference/global-configs/indirect-selection.md:
```diff
@@ -0,0 +1,50 @@
+
+</File>
+
+For example, you can run tests that only refer to selected nodes using an environment variable example:
```
```suggestion
You can run tests that only refer to selected nodes using an environment variable, for example:
```

runleonarun commented on website/docs/reference/global-configs/indirect-selection.md:
```diff
@@ -0,0 +1,50 @@
+
+For example, you can run tests that only refer to selected nodes using `profiles.yml` user configurations:
+
+dbt Core users can configure 
```
```suggestion
```

runleonarun committed:
fix example sentences

runleonarun commented on website/dbt-versions.js:
```diff
@@ -30,6 +30,10 @@ exports.versions = [
 
 exports.versionedPages = [
+  {
+    "page": "/reference/global-configs/indirect-selection",
```
```suggestion
    "page": "reference/global-configs/indirect-selection",
```

runleonarun committed:
Update website/dbt-versions.js

nghi-ly commented on website/docs/docs/core/connect-data-platform/profiles.yml.md:
```diff
@@ -56,4 +57,4 @@ This section identifies the parts of your `profiles.yml` that aren't specific to
 ## User config
 
-You can set default values of global configs for all projects that you run using your local machine. See the docs on [global configs](/reference/global-configs/about-global-configs) for details.
+You can set default values of global configs for all projects that you run using your local machine. Read the docs on [global configs](/reference/global-configs/about-global-configs) for details.
```
sugg: more inclusive word choice for less abled people

```suggestion
You can set default values of global configs for all projects that you run using your local machine. Refer to [About global configs](/reference/global-configs/about-global-configs) for details.
```

nghi-ly commented on website/docs/reference/global-configs/indirect-selection.md:
```diff
@@ -0,0 +1,49 @@
+sidebar: "Indirect selection"
+---
+
+Supply the `--indirect_selection` flag to `dbt test` or `dbt build` to configure which tests will be run for the nodes you specify. You can set this as a CLI flag or an environment variable. In dbt Core, you can also configure user configurations in [YAML selectors](/reference/node-selection/yaml-selectors) or in the `config:` block of `profiles.yml`, which sets default values for all projects.
```
```suggestion
Use the `--indirect_selection` flag to `dbt test` or `dbt build` to configure which tests to run for the nodes you specify. You can set this as a CLI flag or an environment variable. In dbt Core, you can also configure user configurations in [YAML selectors](/reference/node-selection/yaml-selectors) or in the `config:` block of `profiles.yml`, which sets default values for all projects.
```

nghi-ly commented on website/docs/reference/global-configs/indirect-selection.md:
```diff
@@ -0,0 +1,49 @@
+
+Supply the `--indirect_selection` flag to `dbt test` or `dbt build` to configure which tests will be run for the nodes you specify. You can set this as a CLI flag or an environment variable. In dbt Core, you can also configure user configurations in [YAML selectors](/reference/node-selection/yaml-selectors) or in the `config:` block of `profiles.yml`, which sets default values for all projects.
+
+When all flags are set, CLI configurations take precedence, then environment variables, then user configurations. Read the docs on [global configs](/reference/global-configs/about-global-configs) for details.
```
sugg: for improved scannability/readability 

```suggestion
When all flags are set, the order of precedence are as follows. Refer to [About global configs](/reference/global-configs/about-global-configs) for details.

1. CLI configurations
1. Environment variables
1. User configurations
```

nghi-ly commented on website/docs/reference/global-configs/indirect-selection.md:
```diff
@@ -0,0 +1,49 @@
+
+When all flags are set, CLI configurations take precedence, then environment variables, then user configurations. Read the docs on [global configs](/reference/global-configs/about-global-configs) for details.
+
+Options are `empty`, `buildable`, `cautious`, and `eager` (default). By default, dbt indirectly selects all tests if they touch any resource you select. Read more about these options in [Indirect selection in Test selection examples](/reference/node-selection/test-selection-examples?indirect-selection-mode=empty#indirect-selection).
```
```suggestion
You can set the flag to: `empty`, `buildable`, `cautious`, or `eager` (default). By default, dbt indirectly selects all tests if they touch any resource you select. Learn more about these options in [Indirect selection in Test selection examples](/reference/node-selection/test-selection-examples?indirect-selection-mode=empty#indirect-selection).
```

nghi-ly commented on website/docs/reference/global-configs/indirect-selection.md:
```diff
@@ -0,0 +1,49 @@
+
+Options are `empty`, `buildable`, `cautious`, and `eager` (default). By default, dbt indirectly selects all tests if they touch any resource you select. Read more about these options in [Indirect selection in Test selection examples](/reference/node-selection/test-selection-examples?indirect-selection-mode=empty#indirect-selection).
+
+You can run tests that only refer to selected nodes using a CLI configuration, for example:
```
sugg: move "for example" to the front

nghi-ly commented on website/docs/reference/global-configs/indirect-selection.md:
```diff
@@ -0,0 +1,49 @@
+
+</File>
+
+You can run tests that only refer to selected nodes using an environment variable, for example:
```
same sugg

nghi-ly commented on website/docs/reference/global-configs/indirect-selection.md:
```diff
@@ -0,0 +1,49 @@
+
+</File>
+
+You can run tests that only refer to selected nodes using `profiles.yml` user configurations, for example:
```
same sugg

nghi-ly commented on website/docs/reference/global-configs/indirect-selection.md:
````diff
@@ -0,0 +1,49 @@
+
+<File name='Usage'>
+
+```text
````
is this example in a file? should it be `shell` instead of `text` ? i might be misunderstanding tho

nghi-ly reviewed (commented):
looks great! left some comments, none are blocking

runleonarun committed:
Apply suggestions from code review

Co-authored-by: Ly Nguyen <107218380+nghi-ly@users.noreply.github.com>

runleonarun commented on website/docs/reference/global-configs/indirect-selection.md:
```diff
@@ -0,0 +1,53 @@
+
+You can set the flag to: `empty`, `buildable`, `cautious`, or `eager` (default). By default, dbt indirectly selects all tests if they touch any resource you select. Learn more about these options in [Indirect selection in Test selection examples](/reference/node-selection/test-selection-examples?indirect-selection-mode=empty#indirect-selection).
+
+You can run tests that only refer to selected nodes using a CLI configuration, for example:
```
```suggestion
For example, you can run tests that only refer to selected nodes using a CLI configuration:
```

runleonarun commented on website/docs/reference/global-configs/indirect-selection.md:
````diff
@@ -0,0 +1,53 @@
+
+<File name='Usage'>
+
+```text
````
```suggestion
```shell
```

runleonarun commented on website/docs/reference/global-configs/indirect-selection.md:
```diff
@@ -0,0 +1,53 @@
+
+</File>
+
+You can run tests that only refer to selected nodes using an environment variable, for example:
```
```suggestion
For example, you can run tests that only refer to selected nodes using an environment variable
```

runleonarun commented on website/docs/reference/global-configs/indirect-selection.md:
```diff
@@ -0,0 +1,53 @@
+
+</File>
+
+You can run tests that only refer to selected nodes using `profiles.yml` user configurations, for example:
```
```suggestion
For example, you can run tests that only refer to selected nodes using `profiles.yml` user configurations:
```

runleonarun committed:
Apply suggestions from code review

matthewshaver commented on website/docs/reference/global-configs/indirect-selection.md:
```diff
@@ -0,0 +1,53 @@
+
+Use the `--indirect_selection` flag to `dbt test` or `dbt build` to configure which tests to run for the nodes you specify. You can set this as a CLI flag or an environment variable. In dbt Core, you can also configure user configurations in [YAML selectors](/reference/node-selection/yaml-selectors) or in the `config:` block of `profiles.yml`, which sets default values for all projects.
+
+When all flags are set, the order of precedence are as follows. Refer to [About global configs](/reference/global-configs/about-global-configs) for details.
```
```suggestion
When all flags are set, the order of precedence is as follows. Refer to [About global configs](/reference/global-configs/about-global-configs) for more details:
```

matthewshaver commented on website/docs/reference/global-configs/indirect-selection.md:
```diff
@@ -0,0 +1,53 @@
+
+</File>
+
+For example, you can run tests that only refer to selected nodes using an environment variable
```
```suggestion
Or you can run tests that only refer to selected nodes using an environment variable:
```

matthewshaver commented on website/docs/reference/global-configs/indirect-selection.md:
```diff
@@ -0,0 +1,53 @@
+
+</File>
+
+For example, you can run tests that only refer to selected nodes using `profiles.yml` user configurations:
```
Each of these sections starts with "For example" and it feels a little strange to read out loud. Maybe we can vary it a bit?
```suggestion
You can also run tests that only refer to selected nodes using `profiles.yml` user configurations:
```

dbeatty10 replied:
I _think_ is moving to `dbt_project.yml` within https://github.com/dbt-labs/docs.getdbt.com/pull/4814

So we'll want to update to `dbt_project.yml` (after confirming that it actually works, of course).

dbeatty10 commented on website/docs/docs/core/connect-data-platform/profiles.yml.md:
I _think_ is moving to `dbt_project.yml` within https://github.com/dbt-labs/docs.getdbt.com/pull/4814

So we'll either need to merge that one first or merge this one first.

Then deal with any merge conflicts that come up.

runleonarun replied:
@dbeatty10 I think I might change the base branch to #4814 and deal with the merge conflicts now.

dbeatty10 commented on website/docs/reference/global-configs/indirect-selection.md:
```diff
@@ -0,0 +1,53 @@
+sidebar: "Indirect selection"
+---
+
+Use the `--indirect_selection` flag to `dbt test` or `dbt build` to configure which tests to run for the nodes you specify. You can set this as a CLI flag or an environment variable. In dbt Core, you can also configure user configurations in [YAML selectors](/reference/node-selection/yaml-selectors) or in the `config:` block of `profiles.yml`, which sets default values for all projects.
```
I _think_ is moving to `dbt_project.yml` within https://github.com/dbt-labs/docs.getdbt.com/pull/4814

So we'll want to update to `dbt_project.yml` (after confirming that it actually works, of course).

dbeatty10 replied:
```suggestion
Use the `--indirect_selection` flag to `dbt test` or `dbt build` to configure which tests to run for the nodes you specify. You can set this as a CLI flag or an environment variable. In dbt Core, you can also configure user configurations in [YAML selectors](/reference/node-selection/yaml-selectors) or in the `flags:` block of `dbt_project.yml`, which sets project-level flags.
```

dbeatty10 reviewed (commented):
Overall, looks good to me.

Two things:
1. I _think_ this flag is moving to `dbt_project.yml` within [#4814](https://github.com/dbt-labs/docs.getdbt.com/pull/4814), so we'll want to update to `dbt_project.yml` where applicable.
2. We'll need to confirm that it actually works within `dbt_project.yml`, of course.

runleonarun committed:
Update website/docs/reference/global-configs/indirect-selection.md

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

runleonarun committed:
Merge branch 'jerco/flags-behavior-changes' into indirect-selection4773

runleonarun committed:
Merge branch 'jerco/flags-behavior-changes' into indirect-selection4773

runleonarun committed:
Update profiles.yml.md

runleonarun committed:
Apply suggestions from code review

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

runleonarun committed:
adding backindirect selection

runleonarun commented on website/docs/docs/core/connect-data-platform/profiles.yml.md:
```diff
@@ -28,6 +28,7 @@ This section identifies the parts of your `profiles.yml` that aren't specific to
   [debug](/reference/global-configs/logs#log-level): <true | false>
   [version_check](/reference/global-configs/version-compatibility): <true | false>
   [fail_fast](/reference/global-configs/failing-fast): <true | false>
+  [indirect_selection](/reference/global-configs/indirect-selection): <eager | cautious | buildable | empty>
```
We might want to move all of these flags into the project file

dbeatty10 replied:
Good call!

It's covered in #4814 ([preview](https://docs-getdbt-com-git-jerco-flags-behavior-changes-dbt-labs.vercel.app/reference/dbt_project.yml)) with a link to `<global-configs>` like this:

<img width="578" alt="image" src="https://github.com/dbt-labs/docs.getdbt.com/assets/44704949/9b78de52-cac5-4fcb-93ac-f001c9bc1a41">

dbeatty10 commented on website/docs/reference/global-configs/indirect-selection.md:
```diff
@@ -0,0 +1,53 @@
+
+</File>
+
+You can also run tests that only refer to selected nodes using `profiles.yml` user configurations:
```
```suggestion
You can also run tests that only refer to selected nodes using `dbt_project.yml` project-level flags:
```

dbeatty10 commented on website/docs/reference/global-configs/indirect-selection.md:
```diff
@@ -0,0 +1,53 @@
+
+You can also run tests that only refer to selected nodes using `profiles.yml` user configurations:
+
+<File name='profiles.yml'>
```
```suggestion
<File name='dbt_project.yml'>
```

dbeatty10 commented on website/docs/reference/global-configs/indirect-selection.md:
````diff
@@ -0,0 +1,53 @@
+
+```yaml
+
+config:
````
```suggestion
flags:
```

dbeatty10 reviewed (approved):
Left a few suggestions in light of https://github.com/dbt-labs/docs.getdbt.com/pull/4814.

But we can also make these updates later too since user configs will still be a backwards compatible option within v1.8.

runleonarun committed:
Apply suggestions from code review

Co-authored-by: Doug Beatty <44704949+dbeatty10@users.noreply.github.com>

matthewshaver committed:
Updating profiles.yml

matthewshaver committed:
Merge branch 'indirect-selection4773' into profiles-global-configs

matthewshaver committed:
Merge branch 'jerco/flags-behavior-changes' into indirect-selection4773

matthewshaver committed:
Merge branch 'indirect-selection4773' into profiles-global-configs

matthewshaver committed:
Merge branch 'jerco/flags-behavior-changes' into indirect-selection4773

matthewshaver committed:
Merge branch 'indirect-selection4773' into profiles-global-configs

matthewshaver committed:
Profiles global configs (#4992)

## What are you changing in this pull request and why?

Updates the profiles.yml page to reflect the changes to global configs

## Checklist
<!--
Uncomment when publishing docs for a prerelease version of dbt:
- [ ] Add versioning components, as described in [Versioning
Docs](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-entire-pages)
- [ ] Add a note to the prerelease version [Migration
Guide](https://github.com/dbt-labs/docs.getdbt.com/tree/current/website/docs/docs/dbt-versions/core-upgrade)
-->
- [ ] Review the [Content style
guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md)
so my content adheres to these guidelines.
- [ ] For [docs
versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#about-versioning),
review how to [version a whole
page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version)
and [version a block of
content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content).
- [ ] Add a checklist item for anything that needs to happen before this
PR is merged, such as "needs technical review" or "change base branch."

Adding or removing pages (delete if not applicable):
- [ ] Add/remove page in `website/sidebars.js`
- [ ] Provide a unique filename for new pages
- [ ] Add an entry for deleted pages in `website/static/_redirects`
- [ ] Run link testing locally with `npm run build` to update the links
that point to deleted pages
